src/linker/Linker/Tracer.cs

```diff
@@ -90,11 +101,24 @@ bool IsRecordingEnabled ()
 			return recorders != null;
 		}
 
+		bool IsReflectionRecordingEnabled ()
+		{
+			return reflectionPatternRecorders != null;
+		}
+
 		public void AddDirectDependency (object b, object e)
 		{
 			ReportDependency (b, e, false);
 		}
 
+		public void AddDirectDependency (object target, DependencyInfo reason, bool marked)
+		{
+			if (IsRecordingEnabled ()) {
+				foreach (IDependencyRecorder recorder in recorders)
+					recorder.RecordDependency (target, reason, marked);
+			}
+		}
+
 		public void AddDependency (object o, bool marked = false)
```

Comment:
I assume we will eventually remove this method - right? So all the logic around keeping the stack and so on can be removed as well?

Reply:
Do you know if anyone is currently depending on the stack-based tracer output? I wasn't sure if I could replace it with the new tracing because the output will be different. The new tracing should capture a superset of what the old tracing does, but some of the dependencies will show up differently, and I haven't done a full analysis.

Reply:
The only one I know for sure is the analyzer tool in this repo. We would have to adapt that one to the new layout (if it wouldn't work anyway, which is not clear). It's likely we will do some nontrivial amount of work on that tool anyway to produce something usable to hep with reducing the size of trimmed framework.

Reply:
Removed in the latest change.